### PR TITLE
[#191] Add a default database adapter for production environment

### LINKED ISSUE #191: Add a default database adapter for production environment
The rails template is missing the adapter configuration for production in database.yml file
Therefore, I got the `ActiveRecord::AdapterNotSpecified` when building Docker image on production
Screen Shot 2020-07-24 at 17 30 24

### PR DESCRIPTION
**Pull request by malparty**
- Close #191

## What happened 👀

 Add the DATABASE_URL variable in the docker-assets-precompile script

## Insight 📝

All insights are discussed in the issue (#191)
All credits to Hoang for finding a clean fix!!

## Proof Of Work 📹

Tested on the EWA-Payroll project (as we already have the CI set up)

| Before | After |
|---|---|
| image | image |
| image | image |

### CODE REVIEW

**Comment by malparty** [comment]
@hoangmirs @tyrro @Lahphim @junan any comment? :D

**Review by hoangmirs — APPROVED** [review]
LGTM

File: bin/docker-assets-precompile [hunk 1]
```diff
@@ -16,6 +16,8 @@ rails_env = ENV.fetch('RAILS_ENV', 'production')
 if rails_env == 'production'
   env_keys = YAML.load_file('config/application.yml')['docker_build'].keys
   env_keys.each { |name| ENV[name] = 'dummy_value' }
```
Comment (andyduong1920) [inline bin/docker-assets-precompile:18]: The current implementation works but can we modify this function like

```ruby
env_keys = YAML.load_file('config/application.yml')['docker_build'].keys
whitelist_envs = [DATABASE_URL]

env_keys.each do |name|
   if ![whitelist_envs].contain?(env) else
    ENV[name] = 'dummy_value'
   end
end
```

This allow us can extends if we would need to ignore any other new ENV, what do you think?

Comment (malparty) [reply bin/docker-assets-precompile]: cd3d640 ✅ 

Good idea! :)

Comment (andyduong1920) [reply bin/docker-assets-precompile]: My purpose there was we don't need to re-set the `DATABASE_URL` manually and keep the original value for that.

But I just checked the `config/application.yml`['docker_build'] config; seems we don't have the default value for the `DATABASE_URL`, then the `whitelist` here doesn't bring any sense 😂

Another idea, do you think we can create a new scope in the `config/application.yml` that extends from the `docker_build` 🤔 to support precompiling the asset 🤔


```ruby
docker_precompile_asset:
  <<: *docker_build
  DATABASE_URL:  postgres://postgres:postgres@postgres:5432/postgres
```

Or we can set the DATABASE_URL directly in the https://github.com/nimblehq/rails-templates/blob/develop/config/application.yml.tt as it already mentions that is mostly used for the `bin/docker-assets-precompile`

image

Comment (malparty) [reply bin/docker-assets-precompile]: IMO we need to work on a cleaner long-term fix for this workaround of `precompile asset script`. 
So I think we should not change any other file for the sake of cleaning this temporary script 🤔

Comment (malparty) [reply bin/docker-assets-precompile]: There is already [an issue with that](https://github.com/nimblehq/rails-templates/issues/326) and it is not the purpose of this PR (this PR aims to fix a bug in fact ^^)

Comment (andyduong1920) [reply bin/docker-assets-precompile]: Then could you remove the `whitelist_envs` solution above @malparty and put a TODO link to the https://github.com/nimblehq/rails-templates/issues/326 as we will improve that there 🤔

Comment (malparty) [reply bin/docker-assets-precompile:19]: 60189c6

**Review by tyrro — APPROVED** [review]
The changes in `.github/workflows/test_variants.yml` and `Makefil` are also present in https://github.com/nimblehq/rails-templates/pull/351, maybe we can undo them from this PR?

**Review by andyduong1920 — APPROVED** [review]

**Pull request merged**